app/models/rubygem.rb

```diff
@@ -254,6 +254,62 @@ def first_built_date
     versions.by_earliest_built_at.limit(1).last.built_at
   end
 
+  def self.compact_index_versions(date)
+    created_gems = Rubygem.joins(:versions)
+      .select("name, versions.created_at as date, number, platform, info_checksum")
+      .where("versions.created_at > ?", date)
+      .order("versions.created_at, versions.number, platform")
+
+    yanked_gems = Rubygem.joins(:versions)
+      .select("name, versions.yanked_at as date, '-'||number, platform, info_checksum")
+      .where("indexed is false and versions.created_at > ?", date)
+      .order("versions.created_at, versions.number, platform")
+
+    # Arel don't support yet order on unions :(
+    # https://github.com/rails/arel/issues/98
+    sql = "(#{created_gems.to_sql}) UNION (#{yanked_gems.to_sql}) ORDER BY date"
+    gems = Rubygem.find_by_sql(sql)
+
+    gems.map do |gem|
+      CompactIndex::Gem.new(gem.name, [
+        CompactIndex::GemVersion.new(
+          gem.number,
+          gem.platform,
+          gem.info_checksum
+        )
+      ])
+    end
+  end
+
+  def compact_index_info
+    group_by_columns =
+      "number, platform, sha256, info_checksum, ruby_version, rubygems_version, versions.created_at"
+    dep_req_agg =
+      "string_agg(dependencies.requirements, '@' order by rubygems_dependencies.name)"
+    dep_name_agg =
+      "string_agg(rubygems_dependencies.name, ',' order by rubygems_dependencies.name) as dep_name"
+
+    result = Rubygem.includes(versions: { dependencies: :rubygem })
+      .where("rubygems.name = ? and indexed = true and (scope = 'runtime' or scope is null)", name)
+      .group(group_by_columns)
+      .order("versions.created_at, number, platform, dep_name")
+      .pluck("#{group_by_columns}, #{dep_req_agg}, #{dep_name_agg}")
```

Comment:
@segiddins I was fixing this but then I noticed that grop_by_columns is used separated, outside of the aggregation (lines 294 and 296). Is it worth to have two variables that have some content repeated to avoid the interpolation?